src/System.Text.Json/src/System/Text/Json/Document/JsonProperty.cs

```diff
@@ -24,6 +26,21 @@ internal JsonProperty(JsonElement value)
         /// </summary>
         public string Name => Value.GetPropertyName();
 
+        public bool NameEquals(string text)
+        {
+            return NameEquals(text.AsSpan());
```

Comment:
@bartonjs, do we need to add a check like `CheckValidInstance()` to these methods?

Reply:
Yes-ish.  It needs to be done by the helper it defers to, since only it has that knowledge.  (Already commented above)